plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConnectorTest.java

```diff
@@ -0,0 +1,343 @@
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.assertions.Assert.assertEquals;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestSnowflakeConnectorTest
+        extends BaseJdbcConnectorTest
+{
+    protected TestingSnowflakeServer snowflakeServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        snowflakeServer = closeAfterClass(new TestingSnowflakeServer());
+        return createSnowflakeQueryRunner(snowflakeServer, ImmutableMap.of(), ImmutableMap.of(), REQUIRED_TPCH_TABLES);
+    }
+
+    @Override
+    protected SqlExecutor onRemoteDatabase()
+    {
+        return sql -> {
+            try {
+                execute(sql);
+            }
+            catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        };
+    }
+
+    private void execute(String sql)
+            throws SQLException
+    {
+        try (Connection connection = DriverManager.getConnection(snowflakeServer.getJdbcUrl(), snowflakeServer.getProperties());
+                Statement statement = connection.createStatement()) {
+            statement.execute(sql);
+        }
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_AGGREGATION_PUSHDOWN:
+            case SUPPORTS_TOPN_PUSHDOWN:
+            case SUPPORTS_LIMIT_PUSHDOWN:
+                return false;
+            case SUPPORTS_COMMENT_ON_TABLE:
+                return false;
+            case SUPPORTS_ARRAY:
+                return false;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                "tpch.table",
+                "(col_required BIGINT NOT NULL," +
+                        "col_nullable BIGINT," +
+                        "col_default BIGINT DEFAULT 43," +
+                        "col_nonnull_default BIGINT NOT NULL DEFAULT 42," +
+                        "col_required2 BIGINT NOT NULL)");
+    }
+
+    @Override
+    protected TestTable createTableWithUnsupportedColumn()
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                "tpch.test_unsupported_column_present",
+                "(one bigint, two decimal(50,0), three varchar(10))");
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getTrinoTypeName();
+        // TODO: Test fails with type real
```

Comment:
Please leave comment why it fails.

Reply:
Reminder.

Reply:
Complete the error message in the comment.

Reply:
We should leave why it's no result and it's acceptable for skip. The current comment doesn't explain that.

Reply:
I'm sorry for that. I just update the comment and explain the raison why I skipped this type. You can have a look on it please. If you have any idea to figure out the problem,  don't hesitate to tell me. Thanks a lot for your review and help. 😄